setuptools/dist.py

```diff
@@ -52,17 +70,17 @@ def check_importable(dist, attr, value):
         ) from e
 
 
-def assert_string_list(dist, attr, value):
+def assert_string_list(dist, attr: str, value: _OrderedIterable):
```

Comment:
This is more like `assert_ordered_iterable` now. I also realize that whilst this is all technically type-safe, we may not want to allow `dict` for a user-facing reason:
If a user is told they can use a dict here, they're likely going to expect that it means they can use a nested value of some sort, not a "dict with unused values". I also think that forcing a user to coerce their Iterable into a tuple or list in their config file so niche it should be fine.

Reply:
I ended up reverting a lot of this and updating messages instead.